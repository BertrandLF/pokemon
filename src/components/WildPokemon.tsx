import React from 'react';
import './styles/WildPokemon.css';
import Pokemon from './Pokemon';

interface WildPokemonProps {
  pokemon: Pokemon;
  handleOnClick: (id: number) => void;
}

class WildPokemon extends React.Component<WildPokemonProps> {

  render() {
    const pokemon = this.props.pokemon;
    const style = { backgroundImage: `url(${pokemon.sprite})` };
    return <button
      className="wild-pokemon"
      style={style}
      onClick={() => this.props.handleOnClick(pokemon.id)} >
    </button>
  }

}

export default WildPokemon;