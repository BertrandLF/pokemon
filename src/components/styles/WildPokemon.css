.wild-pokemon {
  width: 10em;
  height: 9em;
  margin: 0.7em;
  background-color:rgb(145, 75, 177);
  background-repeat: no-repeat;
  border: none;
}