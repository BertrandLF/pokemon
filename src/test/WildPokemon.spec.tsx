import React from 'react'
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import WildPokemon from '../components/WildPokemon';
import { psyduck } from './PokemonFixture';

it('renders a Pokemon', () => {

  const onClick = () => { };
  const component = shallow(
    <WildPokemon
      pokemon={psyduck}
      handleOnClick={onClick}>
    </WildPokemon>
  )

  expect(toJson(component)).toMatchSnapshot();

})