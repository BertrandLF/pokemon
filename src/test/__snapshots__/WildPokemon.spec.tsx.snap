// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a Pokemon 1`] = `
<button
  className="wild-pokemon"
  onClick={[Function]}
  style={
    Object {
      "backgroundImage": "url(https://someUrl/someImage.png)",
    }
  }
/>
`;
